src/files/BrsFile.ts

```diff
@@ -932,8 +935,9 @@ export class BrsFile {
     }
 
     public getPreviousToken(token: Token) {
-        let idx = this.parser.tokens.indexOf(token);
-        return this.parser.tokens[idx - 1];
+        const parser = this.parser;
```

Comment:
lift the parser out of the loop so we only pay the getter cost once.

Reply:
Could use `_parser` no?

Reply:
No, that's not safe. This is a public method, which provides no guarantees about who will be calling it. Could be a plug-in, in which case _parser might be undefined.